Content.Server/Sound/Components/SpamEmitSoundComponent.cs

```diff
@@ -8,20 +8,36 @@ namespace Content.Server.Sound.Components
     [RegisterComponent]
     public sealed partial class SpamEmitSoundComponent : BaseEmitSoundComponent
     {
-        [DataField("accumulator")]
+        [DataField]
         public float Accumulator = 0f;
 
-        [DataField("rollInterval")]
+        /// <summary>
+        /// Time in seconds between each roll.
+        /// </summary>
+        [DataField]
         public float RollInterval = 2f;
 
-        [DataField("playChance")]
+        /// <summary>
+        /// The maximum amount of extra time (in seconds) that can be
+        /// randomly added to the RollInterval after each roll.
+        /// </summary>
+        [DataField]
+        public float MaxExtraInterval = 2f;
+
+        /// <summary>
+        /// Chance that the sound will play on each roll.
+        /// 0 = never, 1 = always
+        /// </summary>
+        [DataField]
         public float PlayChance = 0.5f;
 
         // Always Pvs.
-        [DataField("popUp")]
+        [DataField]
         public string? PopUp;
 
-        [DataField("enabled")]
+        [DataField]
         public bool Enabled = true;
+
+        public float ExtraInterval;
     }
```

Comment:
If we're adding a max interval you might as well dump the chance because it's functionally equivalent.

Just change the accumulator to a timespan called NextSound and bump it by a random amount between rollinterval and maxextrainterval. I'm also not sure what extrainterval is supposed to be but that looks not required.

Reply:
I'd argue there's a slight difference, since the chance-based effect has the possibility of delaying indefinitely due to RNG. But if that's not the desired behavior, I'm happy to swap it out. The accumulator seems like a messy implementation anyway.

ExtraInterval is just a state variable needed to make this work with the accumulator. It'll go away when refactoring to use a TimeSpan.

Reply:
Technically yeah but indefinitely delaying I think is undesired.